app/models/kudo.rb

```diff
@@ -90,6 +91,25 @@ def guest_cannot_kudos_restricted_work
     end
   end
 
+  def cannot_be_suspended
+    return unless user&.banned || user&.suspended
+
+    commentable = nil
+    if commentable_type == "Work"
+      commentable = Work.find_by(id: commentable_id)
+    elsif commentable_type == "Chapter"
+      commentable = Chapter.find_by(id: commentable_id).work
+    end
+
+    if commentable.nil?
```

Comment:
Instead of adding a new validation in the style of `guest_cannot_kudos_restricted_work`, I think we're close enough to the Rails 6 update that I'd prefer to wait for that, then tackle https://otwarchive.atlassian.net/browse/AO3-5635 first.

Reply:
Okay, I'll put this in draft in the meantime.